src/core/edgeNetwork/requestPayloads/createDataCollectionRequestPayload.js

```diff
@@ -19,7 +19,12 @@ export default () => {
       addIdentity: createAddIdentity(content),
       addEvent(event) {
         content.events = content.events || [];
-        content.events.push(event.toJSON());
+        content.events.push(event);
```

Comment:
This change here probably deserves some explanation. I believe we were calling `event.toJSON()` here because when we log the payload to the console, we don't want the event's methods and stuff showing up on the logged object. Since the time this was written, we've adjusted how we log the payload to the console to be smarter: https://github.com/adobe/alloy/blob/master/src/core/network/injectSendNetworkRequest.js#L24-L39

When we log the payload, we serialize and then deserialize the object. When the object is serialized, `JSON.parse` recursively goes through the payload object, including its events, calling `toJSON`. Because of this, there's no reason to do `event.toJSON()` here anymore. We also don't want to do `event.toJSON()` here anymore because we need access to its methods in the new `getDocumentMayUnload` method below.

Reply:
This also affects when the lastChanceCallback gets called. The cloning of the object is also important so that if the event is queued, when the event is sent, it contains the data as it was when sendEvent was called. We should probably break out some of the toJSON code into a "freeze" method.

Reply:
Thanks for pointing this out. I thought I might be missing something. So, with the changes in this PR, the last chance callback would get called [when the payload gets serialized](https://github.com/adobe/alloy/blob/6658c2625983ae191b63a476277fcfad33185eb1/src/core/network/injectSendNetworkRequest.js#L24), which seems to me to be more appropriate than the way it was previously, since the callback will now be called be as late as possible.

Regarding cloning of customer data, I feel like it should be the responsibility of the data collector to clone the data ASAP, even before it makes its way into the event.

Thoughts on that?

Reply:
Pros of the approach in this PR:
- using the lastChanceCallback to set the consent and preferences on pre-consent events is possible
- Customers have the flexibility of giving an object to sendEvent that can change up until the event is serialized or providing a copy

Cons:
- Customers may be confused about when the xdm and data objects are frozen. (We can solve this with documentation)

I'm sold on the change in this PR.